MO4/Sniffs/Formatting/AlphabeticalUseStatementsSniff.php

```diff
@@ -280,12 +333,82 @@ private function findNewDestination(
             $prevLine      = $tokens[$prevPtr]['line'];
             $prevImportArr = $this->getUseImport($phpcsFile, $prevPtr);
         } while ($prevLine === ($line - 1)
-            && (strcmp($prevImportArr['content'], $import) > 0)
+            && ($this->compareString($prevImportArr['content'], $import) > 0)
         );
 
         return $ptr;
 
     }//end findNewDestination()
 
 
+    /**
+     * Compare namespace strings according defined order function.
+     *
+     * @param string $a first namespace string
+     * @param string $b second namespace string
+     *
+     * @return int
+     */
+    private function compareString($a, $b)
+    {
+        if ('dictionary' === $this->order) {
+            return $this->dictionaryCompare($a, $b);
+        } else if ('string' === $this->order) {
+            return strcmp($a, $b);
+        } else if ('string-locale' === $this->order) {
+            return strcoll($a, $b);
+        } else if ('string-case-insensitive' === $this->order) {
+            return strcasecmp($a, $b);
+        } else {
+            return $this->dictionaryCompare($a, $b);
+        }
+
+    }//end compareString()
+
+
+    /**
+     * Lexicographical namespace string compare.
+     *
+     * Example:
+     *
+     *   use Doctrine\ORM\Query;
+     *   use Doctrine\ORM\Query\Expr;
+     *   use Doctrine\ORM\QueryBuilder;
+     *
+     * @param string $a first namespace string
+     * @param string $b second namespace string
+     *
+     * @return int
+     */
+    private function dictionaryCompare($a, $b)
+    {
+        $min = min(strlen($a), strlen($b));
+
+        for ($i = 0; $i < $min; $i++) {
+            if ($a[$i] === $b[$i]) {
+                continue;
+            }
+
+            if ($a[$i] === self::NAMESPACE_SEPRATOR_STRING) {
+                return -1;
+            }
+
+            if ($b[$i] === self::NAMESPACE_SEPRATOR_STRING) {
+                return 1;
+            }
+
+            if ($a[$i] < $b[$i]) {
+                return -1;
+            }
+
+            if ($a[$i] > $b[$i]) {
+                return 1;
+            }
+        }//end for
+
+        return strcmp(substr($a, $min), substr($b, $min));
```

Comment:
Is it intended that namespace strings are considered equal (return 0) iff the first string is a substring of the second, e.g. "foo\bar\foobar" and "foo\bar" ?